docs/restic.md

```diff
@@ -90,6 +114,67 @@ one of the following commands:
     kubectl -n heptio-ark get podvolumebackups -l ark.heptio.com/backup-name=YOUR_BACKUP_NAME -o yaml
     ```
 
+## Restoring
+
+## Details of Backup & Restore Process with Restic
+
+This section describes in detail how backups/restores with Ark and restic work.
+
+### Backups
+
+- look for annotation on pods
+- ensure a restic repo exists for the namespace
+    - check if a ResticRepository CR exists
+    - if not, create it
+    - RR controller will first check for an existing repo, if not found, init a new one
+- create PodVolumeBackup CRs and wait for them to complete:
+    - PVB controller runs in a daemonset
+    - the controller on the Pod's node will process the PVBs for it
+        - uses a hostPath volume of /var/lib/kubelet/pods to be able to access volume data
+        - finds the path of the volume to backup
+        - runs `restic backup` 
+        - sets status to complete, or failed
+- get the restic snapshot ID for each volume, and annotate the pod *in the backup* with the ID
+
+
+### Restores
+
+- look for snapshot ID annotation on pods
+- ensure a restic repo CR exists for the namespace
+    - check if a ResticRepository CR exists
+    - if not, create it
+    - RR controller will first check for an existing repo, if not found, init a new one
+- add an init container to the pod, whose job is to wait for a done-file to appear for each volume
+- create the pod in the Kube API
+- create PodVolumeRestore CRs
+- PVR controller runs in daemonset
+    - controller on the pod's node will process the PVRs for it
+        - uses a hostPath volume of /var/lib/kubelet/pods to be able to access volume data
+        - waits for pod to be running our init container (so the volumes are mounted)
+        - finds the path of the volume to restore
+        - runs `restic restore`
+        - writes a done file into the volume, in .ark/, whose name is the UID of the Ark restore
+        - sets status to complete (or failed)
+- init container in the Pod:
+    - polls for done file in each volume, in .ark/, with name = UID of the Ark restore
+    - completes
+
+
+## Limitations
+
+There are several limitations that users should be aware of:
+
+- As mentioned previously, you cannot use the main Ark bucket for storing restic backups. We plan to address this
+in a future release.
+- `hostPath` volumes are not supported. [Local persistent volumes][4] are, though!
+- Those of you familiar with [restic][1] may know that it encrypts all of its data. We've decided to use a static, 
+public encryption key for all restic repositories created by Ark. **This means that anyone who has access to your
+bucket can decrypt your restic backup data**. You should ensure that you are limiting access to the restic bucket
+appropriately. We plan to implement full Ark backup encryption in a future release.   
```

Comment:
... full Ark backup encryption, including securing the restic encryption keys, in a future release